src/frontend/src/hooks/UseInstance.tsx

```diff
@@ -1,59 +1,96 @@
 import { useQuery } from '@tanstack/react-query';
-import { useCallback, useState } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 
 import { api } from '../App';
 
+export type InstanceQueryProps = {
+  url: string;
+  pk: string | undefined;
+  params?: any;
+  default: any;
+};
+
 /**
- * Custom hook for loading a single instance of an instance from the API
+ * Custom hook for loading a single endpoint from the API
  *
  * - Queries the API for a single instance of an object, and returns the result.
  * - Provides a callback function to refresh the instance
  *
  * To use this hook:
  * const { instance, refreshInstance } = useInstance(url: string, pk: number)
  */
-export function useInstance(
-  url: string,
-  pk: string | undefined,
-  params: any = {}
-) {
-  const [instance, setInstance] = useState<any>({});
+export function useInstance({
+  url,
+  pk,
+  params = {},
+  defaultValue = {},
+  fetchOnMount = false,
+  hasPrimaryKey = true
+}: {
+  url: string;
```

Comment:
Comment: Using strings as URL references will make supporting #1297 very hard

Reply:
Fair point. I'll make this use your code :)

Reply:
@matmair for reference, a refactor of the URLs - https://github.com/inventree/InvenTree/pull/5686